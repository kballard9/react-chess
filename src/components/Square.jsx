import React from 'react';


const LIGHT_SQUARE = '#f0d9b5';
const DARK_SQUARE = '#b58863';
export const Square = ({black=false, children, onClack}) => (
  <div style={{
    backgroundColor:black? DARK_SQUARE:LIGHT_SQUARE, 
    width:100,
    height:100,
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center'
  }}
  onClick={onClack}>
  { children }
  </div>
)