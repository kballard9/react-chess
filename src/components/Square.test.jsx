import React from 'react';
import TestRenderer from 'react-test-renderer';
import {Square} from './Square';

const LIGHT_SQUARE = '#f0d9b5';

describe('Square', () => {
  it('should render the white square', () => {
    const square = TestRenderer.create(<Square />).toJSON();
    expect(square.props.style.backgroundColor).toEqual(LIGHT_SQUARE);
  })

  it('should render the black square', () => {
    const square = TestRenderer.create(<Square black/>).toJSON();
   
    expect(square).toMatchSnapshot();
  })
})