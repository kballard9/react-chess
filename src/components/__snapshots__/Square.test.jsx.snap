// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Square should render the black square 1`] = `
<div
  style={
    Object {
      "backgroundColor": "#b58863",
      "height": 100,
      "width": 100,
    }
  }
/>
`;
